packages/devtools/test/schema.test.ts

```diff
@@ -786,6 +786,90 @@ describe('schema parsing and compilation', () => {
     })
   })
 
+  it('@immutable directive is supported and properly converted to ICD', () => {
+    expect(
+      createAbstractCompositeDefinition(`
+      type ModelWithImmutableProp @createModel(
+        accountRelation: SINGLE,
+        description: "Test model with an immutable int property"
+      ) {
+        uniqueValue: Int @immutable
+        tag: String! @string(minLength: 1, maxLength: 100)
+      }
+      `),
+    ).toMatchObject({
+      models: {
+        ModelWithImmutableProp: {
+          action: 'create',
+          model: {
+            name: 'ModelWithImmutableProp',
+            immutableFields: ['uniqueValue'],
+            accountRelation: { type: 'single' },
+            description: 'Test model with an immutable int property',
+            schema: {
+              $schema: 'https://json-schema.org/draft/2020-12/schema',
+              type: 'object',
+              properties: { uniqueValue: { type: 'integer' } },
+              additionalProperties: false,
+            },
+          },
+        },
+      },
+    })
+  })
+
+  it('@immutable directive can be set for multiple fields and is supported and properly converted to ICD', () => {
+    expect(
+      createAbstractCompositeDefinition(`
+      type ModelWithImmutableProp @createModel(
+        accountRelation: SINGLE,
+        description: "Test model with an immutable int property"
+      ) {
+        uniqueValue: Int @immutable
+        tag: String! @string(minLength: 1, maxLength: 100)
+        uniqueValue2: Int @immutable
+      }
+      `),
+    ).toMatchObject({
+      models: {
+        ModelWithImmutableProp: {
+          action: 'create',
+          model: {
+            name: 'ModelWithImmutableProp',
+            immutableFields: ['uniqueValue', 'uniqueValue2'],
+            accountRelation: { type: 'single' },
+            description: 'Test model with an immutable int property',
+            schema: {
+              $schema: 'https://json-schema.org/draft/2020-12/schema',
+              type: 'object',
+              properties: { uniqueValue: { type: 'integer' } },
+              additionalProperties: false,
+            },
+          },
+        },
+      },
+    })
+  })
+
+  it('@immutable directive is unsupported in nested objects', () => {
+    createAbstractCompositeDefinition(`
+        interface TestInterface @createModel(description: "Test interface") {
+          test: String! @string(maxLength: 50)
+        }
+      `)
+    expect(() => {
+      createAbstractCompositeDefinition(`
+      interface TestInterface @loadModel(id: "interface ID") {
+        id: ID! @immutable
+      }
+
+      type TestModel @createModel(description: "Test model") {
+        text: String! @string(maxLength: 10)
+      }
+      `)
+    }).toThrow(`Unsupported immutable directive for id on nested object TestInterface`)
```

Comment:
The error message is misleading here, as `TestInterface` is not a nested object.
Also that makes me wonder: has the interface implementation validation logic been updated to make sure the models implementing an interface with immutable fields also set these fields as immutable please?